cpu/BUILD

```diff
@@ -30,6 +30,25 @@ constraint_setting(name = "cpu")
 # Many of the name here are legacy values and probably violate these conditions.
 # We'll try to clean those up over time.
 
+# Special case: Architecture-independent outputs only
+#
+# Some builds are expected to only produce architecture-independent data files,
+# such as configuration files, database seed data, composited images, or even
+# some kinds of interpreted scripts (Shell, Python, Perl, etc).
+#
+# When such a build is being performed, this constraint value may be used to
+# ensure that architecture-dependent builds cannot be performed (except by way
+# of another transition).
+#
+# As a final example, consider building a package of NIC firmware images for
+# many different NIC models.  The package overall is architecture-independent,
```

Comment:
Well, no. It is multi-platform.  The .so files for the different NICs are each architecture dependent and will have their own CPU settings. 

Perhaps this makes sense:
```
cpu = none (or noarch or unset)    for truly nothing
cpu = any (or mixed)  for multi-arch
```
except, that is going to cause confusion. We'll have 2 distinct values which essentially mean "do not use the cpu constraint".    I'm inclined to go with the single `none` or `unset`. To make it more readable in a `select()`.   It's pretty clear what you want if you are selecting on `@platforms//cpu:none`, but if you are selecting on  `cpu:any`, then it reads like a wildcard.

Reply:
lol.   I forgot I said the same thing (but with fewer words) last month.

Reply:
It's not "multi-platform".  On Debian, such a package would be `Architecture: any` and installable on all architectures.

Specifically, the architecture of the host machine containing the NIC is entirely irrelevant to the content of the firmware, which is typically running on a chip that isn't even a supported architecture for the host OS.

IE: The NIC could have a CPU based on the RISC-V architecture, and so the firmware image would be built for a platform that specifies `//cpu:riscv` or whatever (along with some other non-Linux OS).

However, the actual firmware archive itself should be installable on *any* architecture that supports a PCI-E bus; the host machine containing the NIC could be x86_64, ARM, PPC, or anything else.  The same set of firmware bytes work for that specific NIC regardless of what CPU the machine uses.

Furthermore, if this hypothetical Debian package really is being built as `Architecture: any`, then we ought to be able to guarantee that it doesn't actually contain any architecture-specific executables.  It's fine if it contains a preinstall shell-script and the firmware images, but it shouldn't contain an ELF binary because that wouldn't work on arbitrary CPUs.

I think both `//cpu:all` and `//cpu:any` are plausible names for this, but `//cpu:none` implies that there is no CPU at all, which is largely nonsensical and isn't in line with how I think this is likely to be used in practice.

Reply:
Agreed on :all